src/Chirp.Infrastructure/AuthorRepository.cs

```diff
@@ -24,15 +24,17 @@ public IEnumerable<Author> FindAuthorsByEmail(string email)
         return authorCheck;
     }
 
-    public void CreateAuthor(string name, string email)
+    public void CreateAuthor(Author author)
     {
         var authorCheck = _db.Authors
-            .Where(a => a.Name == name && a.Email == email);
+            .Where(a => a.Name == author.Name && a.Email == author.Email);
 
         if (!authorCheck.Any())
         {   
-            int newAuthorId = _db.Authors.Count() == 0 ? 1 : _db.Authors.Max(author => author.AuthorId) + 1;
-            var author = new Author { AuthorId = newAuthorId, Name = name, Email = email };
+            if (author.AuthorId == 0) {
```

Comment:
Could we instead make AuthorId nullable to make it clearer to understand this edge case?

Reply:
ofc I just didn't realize that you could make a nullable int in c#

Reply:
<img width="730" alt="Skærmbillede 2023-11-18 kl  18 46 32" src="https://github.com/ITU-BDSA23-GROUP3/Chirp/assets/133771163/8ce7f4ec-a387-42a0-aaa2-0909a7e7df74">